src/main/java/igym/services/WorkoutService.java

```diff
@@ -84,7 +85,9 @@ public Workout createWorkout(Workout workout, UUID gymId) {
      */
     public List<Workout> getWorkoutsByGymId(UUID gymId) {
         Gym gym = findGymById(gymId);
-        return workoutRepository.findByGym(gym);
+        return workoutRepository.findByGym(gym).stream()
+                .filter(w -> w.getStatus() == Status.active)
+                .toList();
```

Comment:
Here I have two suggestions.

The first one is to retrieve from the repository only **active** workouts, as we are doing in the other entities. 

```suggestion
        return workoutRepository.findByGymAndStatus(gym, Status.active);
```

And the other suggestion is to filter out the inactive exercises here. So then this method will return only active exercises. The strategy that you addopted is filtering the inactive exercises inside the DTO, but I think it should be here, because it is a business rule